js/src/Ice/TcpTransceiver.js

```diff
@@ -2,479 +2,485 @@
 // Copyright (c) ZeroC, Inc. All rights reserved.
 //
 
-const net = require("net");
-
 const Ice = require("../Ice/ModuleRegistry").Ice;
-Ice._ModuleRegistry.require(module,
-    [
-        "../Ice/Debug",
-        "../Ice/SocketOperation",
-        "../Ice/Connection",
-        "../Ice/Exception",
-        "../Ice/Timer"
-    ]);
-
-const Debug = Ice.Debug;
-const SocketOperation = Ice.SocketOperation;
-const Timer = Ice.Timer;
-
-const StateNeedConnect = 0;
-const StateConnectPending = 1;
-const StateProxyConnectRequest = 2;
-const StateProxyConnectRequestPending = 3;
-const StateConnected = 4;
+require("../Ice/Debug");
+require("../Ice/SocketOperation");
+require("../Ice/Connection");
+require("../Ice/Exception");
+require("../Ice/Timer");
 
-//
-// TODO: WORKAROUND: We can directly use Buffer.from once we drop
-// support for Node 4.x
-//
-let createBuffer = null;
-if(Buffer.from)
-{
-    createBuffer = Buffer.from;
-}
-else
-{
-    /* eslint-disable no-buffer-constructor */
-    createBuffer = data => new Buffer(data);
-    /* eslint-enable no-buffer-constructor */
-}
+let TcpTransceiver = {};
 
-class TcpTransceiver
+if (typeof process !== 'undefined')
 {
-    constructor(instance)
+    const net = require("net");
+
+    const Debug = Ice.Debug;
+    const SocketOperation = Ice.SocketOperation;
+    const Timer = Ice.Timer;
+
+    const StateNeedConnect = 0;
+    const StateConnectPending = 1;
+    const StateProxyConnectRequest = 2;
+    const StateProxyConnectRequestPending = 3;
+    const StateConnected = 4;
+
+    //
+    // TODO: WORKAROUND: We can directly use Buffer.from once we drop
+    // support for Node 4.x
+    //
```

Comment:
Do we still support Node 4.x?

Reply:
We can remove this, but it is unrelated to this PR.